polyfactory/utils/types.py

```diff
@@ -9,4 +9,14 @@
 
     NoneType = type(None)  # type: ignore[misc]
 
-__all__ = ("NoneType", "UNION_TYPES")
+
+class Frozendict(dict):
```

Comment:
Why not just use [MappingProxyType](https://docs.python.org/3/library/types.html#types.MappingProxyType)? If you instantiate it with a local copy of the dict, it's effectively a frozen dict.

Reply:
Unfortunately, we don't just need immutability here. We require hashability which MappingProxyType delegates to underling object based on the docs so will still raise an error

Reply:
Then this should include other mutation methods such as `__delitem__`, `update` and `pop`